src/main/scala/scalismo/utils/Conversions.scala

```diff
@@ -153,6 +155,100 @@ object VtkHelpers {
 
 }
 
+object TetrahedronMeshConversion {
+
+  private def extractPointsAndCells(ug: vtkUnstructuredGrid, calculateNewDelaunayTetrahedrons: Boolean = false) = Try {
+    val newUg = if (calculateNewDelaunayTetrahedrons) {
+      val tetrahedralFilter = new vtkDelaunay3D
+      tetrahedralFilter.SetInputData(ug)
+      tetrahedralFilter.Update()
+      tetrahedralFilter.GetOutput()
+    } else ug
+
+    val grids = newUg.GetCells()
+    val numGrids = grids.GetNumberOfCells()
+
+    val points = CommonConversions.vtkConvertPoints[_3D](newUg)
+
+    val idList = new vtkIdList()
+    val cells = for (i <- 0 until numGrids) yield {
+      newUg.GetCellPoints(i, idList)
+      if (idList.GetNumberOfIds() != 4) {
+        throw new Exception("Not a tetrahedral mesh")
+      }
+
+      TetrahedralCell(PointId(idList.GetId(0)), PointId(idList.GetId(1)), PointId(idList.GetId(2)), PointId(idList.GetId(3)))
+    }
+    idList.Delete()
+    (points, cells)
+  }
+
+  private def cleanDoubleOccurrenceOfPoints(
+                                            points: IndexedSeq[Point[_3D]],
+                                            cells: IndexedSeq[TetrahedralCell]
+                                          ): (IndexedSeq[Point[_3D]], IndexedSeq[TetrahedralCell]) = {
+    val cellPointIds = cells.flatMap(_.pointIds).distinct
+    val oldId2newId = cellPointIds.zipWithIndex.map { case (id, index) => (id, PointId(index)) }.toMap
+    val newCells = cells.map(c => TetrahedralCell(oldId2newId(c.ptId1), oldId2newId(c.ptId2), oldId2newId(c.ptId3), oldId2newId(c.ptId4)))
+    val newPoints = cellPointIds.map(id => points(id.id))
+    (newPoints, newCells)
+  }
+
+  def vtkUnstructuredGridToTetrahedralMesh(ug: vtkUnstructuredGrid, recalculateTetrahedralization: Boolean = false): Try[TetrahedralMesh[_3D]] = {
```

Comment:
see my comment above. Do we really ever want to do a different tetrahedralization?

Reply:
I agree - see the answer above